src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -902,6 +933,86 @@ namespace cryptonote
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_notify_tx_pool_inv(int command, NOTIFY_TX_POOL_INV::request& arg, cryptonote_connection_context& context)
+  {
+    MLOG_P2P_MESSAGE("Received NOTIFY_TX_POOL_INV (" << arg.t.size() << " txes)");
+
+    // Create a list to hold transaction hashes missing in our local tx pool.
+    std::vector<crypto::hash> missing_tx_hashes;
+
+    // Iterate over each advertised transaction hash and check our pool and our requested tracker.
+    for (const auto &tx_hash : arg.t)
+    {
+      if (!context.can_process_additional_request())
+      {
+        LOG_ERROR_CCONTEXT("Too many in-flight requests, cannot accept");
+        return 1;
```

Comment:
What's the error return value for this function? It returns 1 on all code paths, is it a logical bug here? Also, it just silently drops all tx hashes once the limit is reached. Maybe it makes sense to queue them for later?